miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSampleController.java

```diff
@@ -689,7 +689,11 @@ public ModelAndView setupForm(@PathVariable Long sampleId, @PathVariable Long pr
 
       model.put("formObj", sample);
       model.put("sample", sample);
-      model.put("sampleTypes", sampleService.listSampleTypes());
+      Collection<String> sampleTypes = sampleService.listSampleTypes();
+      if (!sampleTypes.contains(sample.getSampleType())) {
+        sampleTypes.add(sample.getSampleType());
+      }
+      model.put("sampleTypes", sampleTypes);
       if (LimsUtils.isDetailedSample(sample) && LimsUtils.getIdentityConsentLevel((DetailedSample) sample) == ConsentLevel.REVOKED) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "sample" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)